lib/discourse_code_review/github_category_syncer.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module DiscourseCodeReview
+  module GithubCategorySyncer
+    GithubRepoName = "GitHub Repo Name"
+
+    class << self
+      def ensure_category(repo_name:)
+        Category.transaction(requires_new: true) do
+          category =
+            Category.where(
```

Comment:
Technically you can just do this in one query... no need to run 2. 

`.where('id in (select category_id from category_custom_fields ....)').first`

also... be sure to use `find_by` to save the .first at then end for field lookups.

Reply:
@SamSaffron, here's the clever bit, it's already just doing one query (and it looks very similar to your SQL).

Reply:
I guess if `where` get a relation AREL is sprinkling on even more magic and weaving this stuff together, I am willing to bet this benches slower not that it matter, I guess it is OK